Portfolio Website: Project Details Popup

rmiyoyo opened the pull request:
# Work Done
- Created the cardpopup.js file to implement pop up feature
- Removed projects content section from HTML to implement it dynamically on JavaScript.
- Ran linters tests.

CELYNEBIRUNGI committed:
update cssfile

rmiyoyo committed:
Add mobile menu toggle feature

rmiyoyo committed:
Fix linter errors on style.css

rmiyoyo committed:
Create JavaScript file for implementation of popup feature

CELYNEBIRUNGI committed:
rationalize javascript

rmiyoyo committed:
Remove projects from HTML

rmiyoyo committed:
Implement projects popup in JavaScript

rmiyoyo committed:
Style the project cards in CSS

rmiyoyo committed:
Add popup images for cards

rmiyoyo committed:
Fix linter errors

brainconnect93 commented on style.css:
```diff
@@ -43,6 +45,196 @@
+  }
+}
+
+/* Popup Styling start */
+.model-body {
+  position: relative;
+  width: 320px;
+  height: 90%;
+  margin: 0 auto;
+  padding: 16px;
+  background-color: #fff;
+  border-radius: 16px;
+  z-index: 5;
+}
+
+#model-close {
+  position: absolute;
+  top: 30px;
+  right: 30px;
+}
+
+.line-break {
+  margin: 10px auto;
+  border: solid 1px #ebecf0;
+}
```
- [ ] Hi @rmiyoyo, Great effort working on the pop up modal milestone of the portfolio project. I would like to highlight some of the things that requires attention below, Please kindly work on them

- On `Desktop and mobile view`.
    -  Kindly work on the button alignment and be sure they look exactly like what we have in the Figma design.
    - On class `.model-combine` kindly remove the `justify-content: space-around;` property.
    - and on the class `.model-button`, kindly reduce the `border-radius` to `8px` and add width of `45%` to it
    - lastly kindly give the button some `hover` and `active` effect as seen in Figma design.

- On `mobile view` alone
    - On class `.model-content > p`, kindly increase the `font-size` to `15px` so that it will look like what we have in the Figma design.

| Desktop View  | Mobile View |
| ------------- | ------------- |
| ![sap0](https://user-images.githubusercontent.com/78634841/233502392-d3a716e5-c10f-48a6-b8e8-3371c4c4a7d7.png) | ![sap1](https://user-images.githubusercontent.com/78634841/233502421-9ff81753-a217-4844-b8a7-6eded0806008.png) |

brainconnect93 reviewed (changes requested):
## STATUS: Changes Required :recycle:

### Hi @rmiyoyo, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media.tenor.com/isiLRnxz3zwAAAAM/cat-driving-serious.gif" width="500px" height="300px">

### To Highlight :1st_place_medal: 

-  Descriptive Pull Request title and summary. ✔️
- User clicks the button, the popup with details about the project appears ✔️
- User clicks the close (X) button, the popup disappears ✔️
- Modal popup data is populated with the data from the projects array ✔️
- Pair-Programming activity implemented ✔️
- Good use of Github flow ✔️

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

rmiyoyo committed:
Change technology stack in JS

rmiyoyo committed:
Fix margins and padding issues in popup

rmiyoyo commented:
Hi @brainconnect93, thank you for your input.

rmiyoyo committed:
Fix linter errors

rmiyoyo commented:
# Work Done After First Review
- Worked on button alignment on both mobile and desktop versions.
- Removed ```justify-content: space-around;``` property from ```.model-combine``` class.
- Reduced the ```border-radius``` to 8px and added ```width``` of 45% on the ```.model-button``` class.
- Gave buttons hover and active effects.
- Increased the font-size to 15px in ```.model-content > p``` class.
- Fixed linter errors.

didierganthier commented on cardpopup.js:
```diff
@@ -0,0 +1,290 @@
+  });
+}
+
+body.addEventListener('click', () => {
+  body.style.display = 'none';
+  popUp.remove();
+});
```
[OPTIONAL] Well done on the dynamic popup 💯 but this is `eventListener` although I understand the purpose of adding this (make the popup close when a click is made outside) the popup being part of the `body` causes the popup to also close when any part of it is clicked. 🤔 Removing that line of course eliminates that issue but you also need to add an `eventListener` to the ❌  button to make the popup close when it's clicked, basically the same thing you did for the `body` but with the ❌  `button`. 

Remember, it's very important to have a fully functional app as this is what really showcases your skills and understanding of the module. 🤓

didierganthier commented on cardpopup.js:
```diff
@@ -0,0 +1,290 @@
+      <p>
+      ${projectInfo[1].content}
+      </p>
+      <ul class="tools">
+        <li>${projectInfo[2].tools[0]}</li>
+        <li>${projectInfo[2].tools[1]}</li>
+        <li>${projectInfo[2].tools[2]}</li>
+      </ul>
+      <button class="popBtn show">See project</button>
+    </div>
+  </div>
+  </div>
+
+  <div class="card-body">
+  <div class="card-img swap">
+  <img src="${projectInfo[3].popupPhoto}" width="100%"/>       
+  </div>
+  <div>
+    <h2 class="card-header">${projectInfo[3].header}</h2>
+    <ul class="card-sub">
+      <li class="remove">${projectInfo[1].canopy}</li>
+      <div class="bullet-point"><img src="./images/icons/Counter.png" alt="Bullet Point"></div>
+      <li>${projectInfo[3].backend}</li>
+      <div class="bullet-point"><img src="./images/icons/Counter.png" alt="Bullet Point"></div>
+      <li>${projectInfo[3].year}</li>
+    </ul>
+    <div class="card-content">
+      <p>
+      ${projectInfo[3].content}
+      </p>
+      <ul class="tools">
+        <li>${projectInfo[3].tools[0]}</li>
+        <li>${projectInfo[3].tools[1]}</li>
+        <li>${projectInfo[3].tools[2]}</li>
+      </ul>
+      <button class="popBtn show">See project</button>
+    </div>
+  </div>
+  </div>
+  `;
```
[OPTIONAL] First of all, great job on your code! You've done an excellent job of dynamically creating HTML elements using template literals and object properties. 💯  However, I believe that this could be made a lot easier with a loop. ➿ 

Instead of manually creating the HTML for each project, you could use a loop to iterate over the `projectInfo` array and dynamically create the HTML elements based on each object in the array. This would make your code much more concise and easier to read and maintain.

For example, you could use a `forEach` loop to create the HTML for each project like this:
```
projectInfo.forEach((project) => {
  projects.innerHTML += `
    <div class="card-body">
      <div class="card-img swap">
        <img src="${project.popupPhoto}" width="100%"/>       
      </div>
      <div>
        <h2 class="card-header">${project.header}</h2>
        <ul class="card-sub">
          <li class="remove">${project.canopy}</li>
          <div class="bullet-point"><img src="./images/icons/Counter.png" alt="Bullet Point"></div>
          <li>${project.backend}</li>
          <div class="bullet-point"><img src="./images/icons/Counter.png" alt="Bullet Point"></div>
          <li>${project.year}</li>
        </ul>
        <div class="card-content">
          <p>
            ${project.content}
          </p>
          <ul class="tools">
            <li>${project.tools[0]}</li>
            <li>${project.tools[1]}</li>
            <li>${project.tools[2]}</li>
          </ul>
          <button class="popBtn show">See project</button>
        </div>
      </div>
    </div>
  `;
});
```
This code will iterate over each object in the `projectInfo` array and create the necessary HTML elements for each project. It's a lot shorter and more readable than manually creating the HTML for each project. 📦 

I hope this helps! Let me know if you have any questions. 🤓 ❓

rmiyoyo replied:
Hello @didierganthier, thank you so much for reviewing my project. Also very appreciative for the fact that you went above and beyond to make the suggestions. I am working on integrating them to make my work better. :100:

didierganthier reviewed (approved):
Hi @rmiyoyo and @CELYNEBIRUNGI ,

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

<img src="https://media.giphy.com/media/l9CSJHK88kMTe/giphy.gif" />

## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct GitHub flow ✔️ 

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.

Cheers and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Once again, congratulations on completing this project! 🎉
